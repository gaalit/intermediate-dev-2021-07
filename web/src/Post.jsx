import React from 'react';
import { Card, CardContent, Typography } from '@material-ui/core';

const Post = (props) => {
  console.log('PRROOOPS', props);
  return (
    <Card>
      <CardContent>
        <div>
          <Typography variant="body2" color="textSecondary">
            {props.date}
          </Typography>
          <Typography variant="body1" color="textSecondary">
            {props.author.name}
          </Typography>
          <Typography variant="h5">{props.title}</Typography>
          <Typography variant="body2">{props.body}</Typography>
        </div>
      </CardContent>
    </Card>
  );
};

export default Post;
