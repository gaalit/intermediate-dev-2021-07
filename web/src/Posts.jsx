import React from 'react';
import Post from './Post';

const PostsList = (props) => {
  const { posts } = props;
  console.log('POSTS', posts);

  return (
    <>
      <h3>All Posts:</h3>
      {posts.map((post) => {
        return (
          <Post
            key={post.id}
            title={post.title}
            body={post.body}
            author={post.author}
            date={post.publishedAt}
          />
        );
      })}
    </>
  );
};

export default PostsList;
