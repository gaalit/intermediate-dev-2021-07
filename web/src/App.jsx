import React from 'react';
import { useState, useEffect } from 'react';
import axios from 'axios';
import publishDateReverseOrder from './helperFunctions/publishDateReverseOrder';
// import showUniqueAuthor from './helperFunctions/showUniqueAuthor';
import Posts from './Posts';
import './App.css';

export const App = () => {
  const [posts, setPosts] = useState([]);

  console.log(posts);

  const fetchData = () => {
    axios
      .get('http://localhost:4000/posts')
      .then((res) => {
        setPosts(publishDateReverseOrder(res.data));
      })
      .catch((err) => console.log(err));
  };

  useEffect(() => {
    fetchData();
  }, []);

  return <Posts posts={posts} />;
};
