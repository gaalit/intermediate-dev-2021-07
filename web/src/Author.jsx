import React, { useState } from 'react';
import {
  FormControl,
  RadioGroup,
  FormControlLabel,
  Radio,
} from '@material-ui/core';
import Posts from './Posts';

const Author = ({ name, posts }) => {
  const [value, setValue] = useState(false);

  const handleClick = (event) => {
    if (event.target.value === value) {
      setValue('');
    } else {
      setValue(event.target.value);
    }
  };

  const posted = posts.filter((post) => post.author.name === value);

  return (
    <div>
      <FormControl component="fieldset">
        <RadioGroup aria-label="authors" name="authors" value={value}>
          <FormControlLabel
            value={name}
            control={<Radio onClick={handleClick} />}
            label={name}
          />
        </RadioGroup>
      </FormControl>
      {value ? <Posts posts={posted} /> : ''}
    </div>
  );
};

export default Author;
