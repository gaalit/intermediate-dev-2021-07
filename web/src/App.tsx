import React from 'react';
import logo from './logo.svg';
import { useState, useEffect } from 'react';
import axios from 'axios';

import './App.css';

export function App() {
  const [posts, setPosts] = useState([]);

  const fetchData = () => {
    axios
      .get('http://localhost:4000/posts')
      .then((res) => {
        setPosts(res.data);
      })
      .catch((err) => console.log(err));
  };

  useEffect(() => {
    fetchData();
  }, []);

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}
