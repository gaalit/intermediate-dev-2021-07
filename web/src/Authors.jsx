import React from 'react';
import Author from './Author';

const Authors = ({ authors, posts }) => {
  return (
    <>
      {authors.map((author, index) => {
        return <Author name={author} key={index} posts={posts} />;
      })}
    </>
  );
};

export default Authors;
