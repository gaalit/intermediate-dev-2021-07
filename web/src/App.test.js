/* eslint-env jest */

import React from 'react';
import ReactDOM from 'react-dom';
import { App } from './App';
import Posts from './Posts';
import Authors from './Authors';
import Post from './Post';
import Author from './Author';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

describe('snapshot testing', () => {
  test('snappshot for App component', () => {
    const renderedComponent = renderer.create(<App />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  test('snappshot for Posts component', () => {
    const renderedComponent = renderer.create(<Posts />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  test('snappshot for Authors component', () => {
    const renderedComponent = renderer.create(<Authors />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  test('snappshot for Post component', () => {
    const renderedComponent = renderer.create(<Post />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  test('snappshot for Author component', () => {
    const renderedComponent = renderer.create(<Author />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
