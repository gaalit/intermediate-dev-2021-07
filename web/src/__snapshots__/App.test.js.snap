// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snappshot for App component 1`] = `
Array [
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      View All Posts
    </span>
  </button>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Select by Author
    </span>
  </button>,
]
`;

exports[`snapshot testing snappshot for Author component 1`] = `
<div>
  <fieldset
    className="MuiFormControl-root"
  >
    <div
      aria-label="authors"
      className="MuiFormGroup-root"
      role="radiogroup"
    >
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              checked={false}
              className="PrivateSwitchBase-input-4"
              disabled={false}
              name="authors"
              onChange={[Function]}
              type="radio"
            />
            <div
              className="PrivateRadioButtonIcon-root-5"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </div>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        />
      </label>
    </div>
  </fieldset>
  
</div>
`;

exports[`snapshot testing snappshot for Authors component 1`] = `null`;

exports[`snapshot testing snappshot for Posts component 1`] = `null`;
